### ISSUE TITLE
Yellow card doesn't need Passport to use!

### ISSUE THREAD

**Issue by VCTGomes** [kormax/apple-device-as-access-card#3]
I don't have any passport and, even that, I was able to buy the Yellow card from Alipay for 10 yuans.

![Abrir Jinhua Transit Card cartão](https://github.com/user-attachments/assets/0cd6e44f-956b-4d02-ba4f-7bb7a016803c)

Could you confirm this info?

The yellow card is complete compatible with my Hikvision DS-KV6113-WPE1(B)

Also, I'd like to know if is it possible to replicate same Express Card on Android devices?

**Comment by kormax** [comment]
Didn't know that some cards don't need an ID. Interesting... I've heard that there may be alternative apps which allow the provisioning without any ID too, but haven't looked into it that much.

Also, thanks for sharing the lock model, I think other people may find that useful.

> The yellow card is complete compatible

Technically, all of them use the same T-Union protocol, so it is expected that it should work just like all the others. But good to know they didn't change anything yet.

> same Express Card on Android devices

Sadly, normal Android phones do not allow to set a custom/static UID when performing HCE (as of Android 15, at least), so adding your Android phone "as-is" won't work (some very old phones may use a static UID, but most new ones use random one).

If you have a phone manufactured for the Chinese region (Xiaomi, Huawei, Samsung), some of them have "clone card" functionality in the OEM Wallet app, which allows to clone a Mifare Classic card, including static UID value, which would work in this case.

**Comment by rijckholt** [comment]
## Hi everyone,  

I wanted to share my experience and some additional information regarding the process of setting up a Chinese subway card via Alipay and whether verification is necessary.  

1. I was able to install Alipay, but it took quite some time to get my account verified.  
2. The only verification method initially offered in the app required linking a Chinese mainland bank card, which I obviously do not have.  
3. After reaching out to the helpdesk through the app (without receiving a direct reply) and logging into the Alipay Chinese website, I was eventually presented with the option to verify my account by taking a photo of myself. I had already uploaded my passport at this stage.  
4. It’s unclear whether starting the verification process on the web or my earlier helpdesk interaction triggered the photo ID verification flow.  

### After successfully completing the verification process:  
- I was able to install the metro card mentioned in the guide and top it up.  
- I added the card to my wallet on my Apple Watch.  
- I can now open my door (controlled by Unifi Access) by holding my watch close to the reader.  
- It works flawlessly every time and is incredibly convenient.  

### My neighbor's experience:  
However, my neighbor had a different experience. While he followed the guide, he was unable to complete the Alipay identity verification process, as the app did not provide him with an alternative verification option. After reading this post/issue on GitHub, I suggested he try finding the Yellow Card under the NFC app section in Alipay.  

To his surprise, he was able to:  
- Locate and install the Yellow Card.  
- Top it up with 10 Yuan using a credit card already linked to his Alipay account.  
- Add the card to Apple Wallet when prompted by his iPhone.  
- Enable Express Mode, allowing him to open his door (controlled by Unifi Access) by simply presenting his locked phone to the reader.  

### Testing on my wife’s iPhone:  
Curious to see if this method worked in other scenarios, I tested it on my wife’s iPhone:  
- I installed Alipay, verified her phone number, and linked her credit card to the account.  
- I searched for the NFC app, found the Yellow Card, installed it, and topped it up with 10 Yuan.  
- I successfully added the card to her Apple Wallet and configured it for our Unifi Access system.

**Comment by diepnt90** [comment]
We can pay Pasmo, Suica of Japan and they work. Just a little min pay expensive than China

**Comment by kormax** [comment]
> We can pay Pasmo, Suica of Japan and they work. Just a little min pay expensive than China

You're right, FeliCa cards have a static IDM values between reads, unless the card is re-added.

That said, there are much more readers supporting Type A/B UID matching vs F IDM matching. FeliCa trick would be useful for JP region primarily.

**Comment by fudgebucket27** [comment]
Thanks for the tip for using yellow card, works for me. Able to unlock a Carbine CEL 3 in 1 digital lock with it now.

**Comment by miklosbard** [comment]
for me Suica doesn’t work. I have been waiting for this for 3 years. Thanks guys, this is amazing

**Comment by jhonnybonny** [comment]
Transport cards PASMO (felica) can be added without balance (u don't need pay)
Just download application from the app store

**Comment by f1shy-dev** [comment]
wait what!!! it worked. now i don't know if this is fixed UID or not; will have to test

![Image](https://github.com/user-attachments/assets/d7414640-7d96-4795-8b94-ec8c01ff2293)

**Comment by apexmod2** [comment]
> wait what!!! it worked. now i don't know if this is fixed UID or not; will have to test

Just tried this with the Ubiquiti's G4 Doorbell Pro's NFC scanner and Apple Wallet sadly it seems to be randomized.

**Comment by jhonnybonny** [comment]
> > wait what!!! it worked. now i don't know if this is fixed UID or not; will have to test
> 
> Just tried this with the Ubiquiti's G4 Doorbell Pro's NFC scanner and Apple Wallet sadly it seems to be randomized.

UID - STATIC

**Comment by kormax** [comment]
> > wait what!!! it worked. now i don't know if this is fixed UID or not; will have to test
> 
> Just tried this with the Ubiquiti's G4 Doorbell Pro's NFC scanner and Apple Wallet sadly it seems to be randomized.

I’ve mentioned this here before, but most NFC readers, including UniFi, do not attempt to interpret IDM values for detected FeliCa credentials as usable identifiers, so they cannot be used.

As for why it detects a random identifier when you try the enrollment - because when emulating FeliCa after manual authentication, your Apple device also exposes the “Value Added Services” applet, which uses NFC A with a random UID.

The only FeliCa card type this doesn’t happen to is Octopus, as it suppresses VAS, ~~but I don’t think it’s gonna change anything in your case~~.
***UPDATE***: I've verified that Octopus cards work with UniFi access, as there's no NFC-A/F confusion due to VAS being disabled, and FeliCa IDM is used instead.

**Comment by f1shy-dev** [comment]
Can confirm the Felica is a rotating UID but the Chinese gold T-Union card is a fixed UID! It only cost me like 10 Yuan ($1.50) to get.

**Comment by piotr-siedlak** [comment]
I check it 5 min ago and can't add any card other that Yellow Card but my ID tag is randomized. Any solution ?

**Comment by kormax** [comment]
Just checked - all of my Chinese cards still have a static UID.

**Comment by piotr-siedlak** [comment]
iOS 18.5 
interesting after I remove card and add it again it starts working...

**Comment by JacktheRanger** [comment]
The Jinhua T-Union (gold) card does not work on my iPhone 16 PM, iOS 26 Beta. My doorlock does not react to it. Flipper zero shows "MOT_T_EP No data was parsed".

**Comment by kormax** [comment]
> The Jinhua T-Union (gold) card does not work

Which reader?

**Comment by diepnt90** [comment]
> The Jinhua T-Union (gold) card does not work on my iPhone 16 PM, iOS 26 Beta. My doorlock does not react to it. Flipper zero shows "MOT_T_EP No data was parsed".

Try ios 18 first and it works?

**Comment by JacktheRanger** [comment]
> > The Jinhua T-Union (gold) card does not work
> 
> Which reader?

I actually got two different smart door locks: one from Kaadas, the other from an unbranded company. I don’t think it’s the door locks’ fault, as the Jinhua T-Union card (along with Jinan and Qingdao T-Union cards) all have an SAK value of 20. I also tried the PASMO card, but with no luck.. I suspect it has something to do with iOS 26. Or they just fixed the loophole for newly issued cards. Maybe someone with an iOS26 device can test it-with their old card and a newly issued card, if the old one still works, that is.

**Comment by ICURSEDANGEL** [comment]
Is there any other card that doesn't require Alipay to work? It's not accepting my cards and to get a tourist card my Syrian passport is not accepted either, have just tried PASMO but the UID changes and my flipper zero also confirmed it sadly.

**Comment by tigeryfan** [comment]
> The Jinhua T-Union (gold) card does not work on my iPhone 16 PM, iOS 26 Beta. My doorlock does not react to it. Flipper zero shows "MOT_T_EP No data was parsed".

Try using Extra Actions → Read Specific Card Type → and try the first few. The Flipper is trying to read the card as an EMV card, but they still have UIDs

**Comment by JacktheRanger** [comment]
> > The Jinhua T-Union (gold) card does not work on my iPhone 16 PM, iOS 26 Beta. My doorlock does not react to it. Flipper zero shows "MOT_T_EP No data was parsed".
> 
> Try using Extra Actions → Read Specific Card Type → and try the first few. The Flipper is trying to read the card as an EMV card, but they still have UIDs

Nah..It doesn't matter. I used my Proxmark3 RDR4 later on... The problem is that the SAK and ATQA are just wrong. I don't think Apple issues new Mifare transport cards anymore (at least not on iOS 26).

**Comment by kormax** [comment]
Cards marked with T-Union logo in Wallet were never Mifare. They are using a ISO7816-filesystem-based T-Union protocol.

Even the original ones issued for Beijing and Shanghai in IOS 11-13 were seemingly based on "Fudan Micro" protocol, as was indicated by product names found on Apple servers and iOS.

**Comment by kormax** [comment]
If an access system is programmed to expect a specific SAK and ATQA, like the one used on Mifare cards - I'm afraid there isn't much one could do.

**Comment by JacktheRanger** [comment]
> Cards marked with T-Union logo in Wallet were never Mifare. They are using a ISO7816-filesystem-based T-Union protocol.
> 
> Even the original ones issued for Beijing and Shanghai in IOS 11-13 were seemingly based on "Fudan Micro" protocol, as was indicated by product names found on Apple servers and iOS.

I can’t verify this because I don’t have any iOS 11-13 devices.
But I’m curious about the “Fudan Micro” protocol, because I haven’t heard about that before, but Fudan Microelectronics Group is known for producing enhanced Mifare Cards (They call them 3rd gen loophole-free cards in China, aka FM11RF08S). If they are using a unique protocol, do you happen to know the SAK and ATQA or other characteristics of those cards?


> If an access system is programmed to expect a specific SAK and ATQA, like the one used on Mifare cards - I'm afraid there isn't much one could do.

I found that funny that some door locks accept anything other than the SAK and ATQA of the original Mifare card, since those are verified at the very beginning, before AUTH-A, nt, nr, ar are exchanged. The door lock I have doesn’t even accept Mifare cards emulated on Android phones because the interaction latency is too high during the reader’s SELECT after UID step😅

**Comment by kormax** [comment]
It would seem weird to me if Apple, a partner of NXP, who already have access to official Mifare applets, would use a clone made by Fudan.

I'm more inclined to believe that "Fudan Micro" applet is emulating some sort of FMCOS or CPU cards, which use ISO7816.

But yeah, we have no way of checking that now. At least for people outside of China.
A Chinese person could check a thing though: 
Bejing has a "RuubyPay" card, which is still issuable through an app, and it is marked as Fudan Micro on Apple's servers.
It would indeed be very interesting to look at what characteristics it possesses in comparison to T-Union.

**Comment by tigeryfan** [comment]
I don't think Beijing offers "RuubyPay" (亿通行) cards anymore - they've switched to T-Union "Yikatong" (一卡通) cards. Old cards still work, but they aren't issued anymore either through Wallet or the Yikatong app. Right now, the RuubyPay app only issues QR-Code tickets.

**Comment by MinhThienDX** [comment]
> I actually got two different smart door locks: one from Kaadas, the other from an unbranded company. I don’t think it’s the door locks’ fault, as the Jinhua T-Union card (along with Jinan and Qingdao T-Union cards) all have an SAK value of 20. I also tried the PASMO card, but with no luck.. I suspect it has something to do with iOS 26. Or they just fixed the loophole for newly issued cards. Maybe someone with an iOS26 device can test it-with their old card and a newly issued card, if the old one still works, that is.

There is no loophole to be fixed actually.
It's legacy so less secure, you can read more here:
https://www.passninja.com/tutorials/nfc-protocols/how-to-make-an-nfc-pass-scan-as-static-uid

I got my noname lock to work with the Jinhua card using Apple Watch on watchOS 26.
I bought it on Alipay on iOS 26 then add to Apple Wallet then transfer to watchOS 26.
I tried Navigo card for €1 but it is not static.
You can check the card using an android phone with nfc app (the serial number and ATS must be the same for multiple swipes).

The unlock speed is meh though.
I initially think it is as fast as using card but I must hold the watch against the lock for about 5 seconds.
But my lock is old, cheap, noname and purchased on 2022 so YMMV.

A little comparison sort by modernity:
| Method | Unlock speed | Convenience | Cost | Disadvantages |
|----------|----------------|----------------|------|------------------|
| Key | Fast, very slow if drunk | Must take out of pocket | $ | Must bring along<br/>Not secure |
| Card | Very fast | Must take out of pocket | $$ | Must bring along |
| Fingerprint | Very fast | Always ready | | Hard to use when rain |
| Phone app | Very slow | Must take out of pocket | Already paid | Must bring along<br/>Depend on app and network condition<br/>Hard to use when rain |
| Watch app | Very slow | Took a few taps | $$$ | Not always available as an app<br/>Depend on app and network condition<br/>Hard to use when rain |
| Phone nfc | Depend on lock | Must take out of pocket | Already paid | Must bring along<br/>Hard or impossible to setup, android is much easier |
| Watch nfc | Depend on lock | Already on wrist | $$$ | Hard or impossible to setup, android is much easier |

**Comment by ma-tecik** [comment]
Hi. Firstly thanks a lot for all the info all of you share until now. I've a problem, has someone an idea and help me please? I can't get the yellow card, it stuck on loading loop. I don't have a passport and don't think about visiting abroad soon, so I can't verify my account.
